subsys/bluetooth/services/fast_pair/fp_advertising.c

```diff
@@ -95,6 +95,10 @@ static size_t bt_fast_pair_adv_data_size_discoverable(void)
 
 size_t bt_fast_pair_adv_data_size(struct bt_fast_pair_adv_config fp_adv_config)
 {
+	/* It is assumed that this function executes in the cooperative thread context. */
+	__ASSERT_NO_MSG(!k_is_preempt_thread());
+	__ASSERT_NO_MSG(!k_is_in_isr());
```

Comment:
Also add a note to the `bt_fast_pair_adv_data_size` API docs that the function needs to be called in the COOP priority